src/app/components/todolist/todolist.component.ts

```diff
@@ -55,7 +48,16 @@ export class TodolistComponent implements OnInit {
   deleteTodo(id: number): void {
     // filters on the list of todos in the arry and looks for the matching id we pass in
     this.todos = this.todos.filter(todo => todo.id != id);
+  }
+
+  // filters on the list of todos and looks for ones that do not have the boolean completed as true
+  todoRemaining(): number {
+    return this.todos.filter(todo => !todo.completed).length;
+  }
 
+  checkAllTodos(): void {
```

Comment:
Could we not have the following in the UI, then all that would 

```
 <input type="checkbox" [checked]="todo.completed">
```

Wouldn't that solve this problem?

Reply:
Changing to [checked] rather than the ngModel means that the class 'todo-item-label' doesn't get the '.completed' added to it from the ngClass.  This means that no strike through goes through the item.